validation.sh

```diff
@@ -11,4 +11,8 @@
 set -e
 
 DIR="$(dirname "$0")"
-"$DIR/run.sh" "$DIR/src/run_validation.py" $@
+if [  -n "$(uname -a | grep Ubuntu)" ]; then
+    "sudo" "$DIR/run.sh" "$DIR/src/run_validation.py" $@
```

Comment:
Just completely remove the sudo part from this code.
Simply echo this line so inform user that if their user account does not have access to docker, they need to run the workflow in sudo.

This is not a problem in our docker image as it already assume the runner socket, and the runner would not have this issue because their user is already added to the docker group.

See RPM/DEB example (just for reference): https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/integ_test/distribution_rpm.py#L30

Reply:
Testing without sudo and just adding the user to docker group:
```

Running ./src/run_validation.py --version 1.3.8 --distribution docker --os_build_number 7019 --osd_build_number 5411 ...
CONTAINER ID   NAME      CPU %     MEM USAGE / LIMIT   MEM %     NET I/O   BLOCK I/O   PIDS

Docker started
2023-02-03 23:05:23 INFO     docker_daemon is running
2023-02-03 23:05:23 INFO     Image does not exist at local, proceed with Pull from the dockerhub
2023-02-03 23:05:23 INFO     Pulling opensearchproject/opensearch:1.3.8 from the dockerhub
2023-02-03 23:05:32 INFO     Image is pulled at local : sha256:4460662780001ed00650637d774570dd3cfb64be1afc1461b6620fc6dc625240
 : opensearchproject/opensearch:1.3.8
2023-02-03 23:05:33 INFO     Image does not exist at local, proceed with Pull from the dockerhub
2023-02-03 23:05:33 INFO     Pulling opensearchproject/opensearch-dashboards:1.3.8 from the dockerhub
2023-02-03 23:05:46 INFO     Image is pulled at local : sha256:96d6ab746fd71a16b6f984e101ece24158c030b51bcfff0375d393cb39ff4014
 : opensearchproject/opensearch-dashboards:1.3.8
2023-02-03 23:05:46 INFO     the OS image ID is : sha256:4460662780001ed00650637d774570dd3cfb64be1afc1461b6620fc6dc625240

2023-02-03 23:05:46 INFO     the OSD image ID is : sha256:96d6ab746fd71a16b6f984e101ece24158c030b51bcfff0375d393cb39ff4014



2023-02-03 23:05:46 INFO     Fetching access token from dockerHub
2023-02-03 23:05:46 INFO     Fetching mainfest from DockerHub
2023-02-03 23:05:47 INFO     sha256:4460662780001ed00650637d774570dd3cfb64be1afc1461b6620fc6dc625240 --> DockerHub image digest opensearchstaging/opensearch:1.3.8.7019
2023-02-03 23:05:47 INFO     Fetching mainfest from local
2023-02-03 23:05:47 INFO     sha256:4460662780001ed00650637d774570dd3cfb64be1afc1461b6620fc6dc625240 --> local image digest opensearchproject/opensearch:1.3.8
2023-02-03 23:05:47 INFO     Fetching access token from dockerHub
2023-02-03 23:05:47 INFO     Fetching mainfest from DockerHub
2023-02-03 23:05:47 INFO     sha256:96d6ab746fd71a16b6f984e101ece24158c030b51bcfff0375d393cb39ff4014 --> DockerHub image digest opensearchstaging/opensearch-dashboards:1.3.8.5411
2023-02-03 23:05:47 INFO     Fetching mainfest from local
2023-02-03 23:05:47 INFO     sha256:96d6ab746fd71a16b6f984e101ece24158c030b51bcfff0375d393cb39ff4014 --> local image digest opensearchproject/opensearch-dashboards:1.3.8
2023-02-03 23:05:47 INFO     Image digest is validated


Changing "OS_IMAGE_PLACEHOLDER" to "opensearchproject/opensearch:1.3.8" in /tmp/tmp42fa8n__/docker-compose.yml
Changing "OSD_IMAGE_PLACEHOLDER" to "opensearchproject/opensearch-dashboards:1.3.8" in /tmp/tmp42fa8n__/docker-compose.yml
Changing "OS1_NAME" to "opensearch-node1-test" in /tmp/tmp42fa8n__/docker-compose.yml
Changing "OS2_NAME" to "opensearch-node2-test" in /tmp/tmp42fa8n__/docker-compose.yml
Changing "OSD_NAME_PLACEHOLDER" to "opensearch-dashboards-test" in /tmp/tmp42fa8n__/docker-compose.yml
2023-02-03 23:05:47 INFO     final test_result = False
```

Reply:
Thank you, @peterzhuamazon , let us update the script accordingly.

Reply:
You dont need to update the script, just ask user to make sure docker can run, and they would do the necessary later.